### PR TITLE
earthly: add missing buildFlags and tags

### LINKED ISSUE #125463: earthly is missing some compile time flags
**Describe the bug**

1. `earthly --verrsion` prints some information that misses the actual version
2. `earhly +$target` can't do anything as it doesn't know what buildkit image to use

**Expected behavior**

1. The version is properly printed
2. `earthly` knows the default image

**Additional context**

earthly/earthly#1025

**Notify maintainers**

@Mdsp9070
@kalbasit

**Metadata**

```plain
$ nix-shell -p nix-info --run "nix-info -m" -I nixpkgs=channel:nixos-unstable
 - system: `0`
 - host os: `Linux 5.4.119, NixOS, 20.09.20210530.eae0cab (Nightingale)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210531_e8f585b`
 - channels(root): `""`
 - channels(aroemer): `"nixpkgs-20.09.4025.dc326c78a93"`
 - channels(nmelzer): `""`
 - nixpkgs: `0`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- earthly
```

### PR DESCRIPTION
**Pull request by Artturin**
###### Things done

closes #125463
@NobbZ  can you test this since i haven't used earthly

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NobbZ** [comment]
This does not work yet, I have an idea why, but can only check when at home *and* kids are sleeping.

If someone else has time and opportunity to dig into this, please remove the quotes around the image name, rebuild and run an arbitrary example image.

Even `earthly +foo` should "work". It is fine if it doesn't error because of an invalid reference format.

Just paste error here.

**Comment by Artturin** [comment]
works now 

```
earthly github.com/earthly/hello-world:main+hello
           buildkitd | Starting buildkit daemon as a docker container (earthly-buildkitd)...
      buildkitd-pull | Pulling buildkitd image...
      buildkitd-pull | ...Done
           buildkitd | ...Done
             ongoing | internal (5 seconds ago)
             context | --> git context github.com/earthly/hello-world:main+hello
      busybox:1.32.0 | --> Load metadata linux/amd64
g/e/hello-world:main+base | --> FROM busybox:1.32.0
g/e/hello-world:main+base | [          ] resolve docker.io/library/busybox:1.32.0@sha256:bde48e1751173b709090c2539fdf12d6ba64e88ec7a4301591227ce925f3c678 ... 0%         [██████████] resolve docker.io/library/busybox:1.32.0@sha256:bde48e1751173b709090c2539fdf12d6ba64e88ec7a4301591227ce925f3c678 ... 100%
g/e/hello-world:main+base | [██████████] sha256:ea97eb0eb3ec0bbe00d90be500431050af63c31dc0706c2e8fb53e16cff7761f ... 100%
g/e/hello-world:main+base | [██████████] extracting sha256:ea97eb0eb3ec0bbe00d90be500431050af63c31dc0706c2e8fb53e16cff7761f ... 100%
g/e/hello-world:main+hello | --> RUN echo 'Hello, world!'
g/e/hello-world:main+hello | Hello, world!
              output | --> exporting outputs
              output | [██████████] sending tarballs ... 100%
=================================== SUCCESS ====================================
```

**Review by NobbZ — APPROVED** [review]

**Review by vladaionescu — APPROVED** [review]
Looks good to me too, as far as I can tell.

Thanks for maintaining Earthly as a package!

File: pkgs/development/tools/earthly/default.nix [hunk 1]
```diff
@@ -13,6 +13,18 @@ buildGoModule rec {
 
   vendorSha256 = "sha256-q3dDV0eop2NxXHFrlppWsZrO2Hz1q5xhs1DnB6PvG9g=";
 
+  buildFlagsArray = ''
+    -ldflags=
```
Comment (SuperSandro2000) [inline pkgs/development/tools/earthly/default.nix:17]:
```suggestion
    -ldflags=
      -s -w
```

Comment (Artturin) [reply pkgs/development/tools/earthly/default.nix:17]: what do they do?

Comment (Artturin) [reply pkgs/development/tools/earthly/default.nix:17]: ah shrinking
https://lukeeckley.com/post/useful-go-build-flags/

`We can omit the symbol table, debug information and the DWARF table with the following flags: -s -w`

**Comment by Artturin** [comment]
builds and and `earthly github.com/earthly/hello-world:main+hello` runs

**Review by zoedsoupe — APPROVED** [review]

**Pull request merged**